lua/arrow/ui.lua

```diff
@@ -220,8 +222,8 @@ end
 local function createMenuBuffer(filename)
 	local buf = vim.api.nvim_create_buf(false, true)
 
-	vim.b.filename = filename
-	vim.b.arrow_current_mode = ""
+	vim.b[buf].filename = filename
+	vim.b[buf].arrow_current_mode = ""
```

Comment:
the buffer local value should be new created menu buf, otherwise these option is set in previous buf, cause curren_index to stay unchanged during deleting